### PR TITLE
ET-3537 proxy api calls

### PR DESCRIPTION
**Pull request by Robert-Steiner**
## What 🕵️ 🔍

- replaces the openapi integration with a vpc link proxy
- removes the openapi template and python script
- [Guide](https://docs.aws.amazon.com/apigateway/latest/developerguide/setup-http-integrations.html)

## References 📝 🔗

- [Jira task](https://xainag.atlassian.net/browse/ET-3537)

## Reviewer checklist

*Reviewer agreement:*

* Reviewers shouldn't review a "Draft Pull Request", but rather wait for it to be marked as "Ready for Review".
* Reviewers do **not** commit or merge the pull request.
* If/when there are changes that need to be applied (applying Terraform or deploying Kubernetes manifests), the **assignee** is the one responsible for doing it.
* Use the labels `not-applied`, `partially-applied` and `fully-applied` so others know the current progress.
  * Remember to change labels accordingly once you start and finish to apply the changes.

### CODE REVIEW

File: aws/stage/api_gateway/main.tf [hunk 1]
```diff
@@ -8,32 +8,71 @@ data "terraform_remote_state" "lambdas" {
...
-    token_name : "authorizationToken",
-    authorizer_credentials : aws_iam_role.authorizer_invocation.arn
-  })
+resource "aws_api_gateway_rest_api" "tenant" {
+  name        = "api_${var.tenant}"
+  description = "API for ${var.tenant}"
 
   endpoint_configuration {
     types = ["REGIONAL"]
   }
```
Comment (joecks) [inline aws/stage/api_gateway/main.tf:17]: ❓ based on
```
When API requests predominantly originate from an EC2 instance or services within the same region as the API is deployed, a regional API endpoint will typically lower the latency of connections and is recommended for such scenarios.
```
https://docs.aws.amazon.com/apigateway/latest/developerguide/create-regional-api.html

It seems that REGIONAL is ment for services that are calles from with-in the same region, but in our case we will call the API Gateway from external clients.

I moght be wrong but wouldn't then EDGE make more sense.

File: aws/stage/api_gateway/main.tf [hunk 2]
```diff
@@ -8,32 +8,71 @@ data "terraform_remote_state" "lambdas" {
...
+  connection_type         = "VPC_LINK"
+  connection_id           = aws_api_gateway_vpc_link.this.id
+  request_parameters = {
+    "integration.request.path.proxy" : "method.request.path.proxy"
+  }
+}
+
+resource "aws_api_gateway_deployment" "tenant" {
+  depends_on  = [aws_api_gateway_method.proxy]
+  rest_api_id = aws_api_gateway_rest_api.tenant.id
```
Comment (joecks) [inline aws/stage/api_gateway/main.tf:67]: ❓ Shouldn't that block be moved above "aws_api_gateway_stage"

**Review by joecks — COMMENTED** [review]
Lets have a call I have some questions :)

File: aws/stage/api_gateway/main.tf (hunk 2 above)
Comment (Robert-Steiner) [reply aws/stage/api_gateway/main.tf:67]: we can move it above aws_api_gateway_stage. but it is not syntactically necessary. terraform reads all tf files that are in the cwd and creates a graph from them

**Review by joecks — APPROVED** [review]
LGTM

**Pull request merged**